plugin.video.youtube/resources/lib/youtube_plugin/kodion/impl/xbmc/xbmc_items.py

```diff
@@ -7,6 +7,75 @@
 from ...items import VideoItem, AudioItem, UriItem
 from ... import utils
 from . import info_labels
+from ...items import utils as item_utils
+
+
+def to_play_item(context, play_item):
+    context.log_debug('Converting PlayItem |%s|' % play_item.get_uri())
+
+    major_version = context.get_system_version().get_version()[0]
+    thumb = play_item.get_image() if play_item.get_image() else u'DefaultVideo.png'
+    title = play_item.get_title() if play_item.get_title() else play_item.get_name()
+    fanart = ''
+    settings = context.get_settings()
+    if major_version > 17:
+        list_item = xbmcgui.ListItem(label=utils.to_unicode(title), offscreen=True)
+    else:
+        list_item = xbmcgui.ListItem(label=utils.to_unicode(title))
+
+    list_item.setProperty(u'IsPlayable', u'true')
+
+    if play_item.get_fanart() and settings.show_fanart():
+        fanart = play_item.get_fanart()
+    if major_version <= 12:
+        list_item.setIconImage(thumb)
+        list_item.setProperty("Fanart_Image", fanart)
+    elif major_version <= 15:
+        list_item.setArt({'thumb': thumb, 'fanart': fanart})
+        list_item.setIconImage(thumb)
```

Comment:
replace by `icon`:`xxx` above

Reply:
icon wasn't added to setArt until Kodi 16